internal/scheduler/scheduler.go

```diff
@@ -0,0 +1,76 @@
+package scheduler
+
+import (
+	deploymentskyscannernetv1alpha1 "github.com/Skyscanner/argocd-progressive-rollout/api/v1alpha1"
+	"github.com/Skyscanner/argocd-progressive-rollout/internal/utils"
+	argov1alpha1 "github.com/argoproj/argo-cd/pkg/apis/application/v1alpha1"
+	"github.com/argoproj/gitops-engine/pkg/health"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// Scheduler returns a list of apps to sync for a given stage
+func Scheduler(apps []argov1alpha1.Application, stage deploymentskyscannernetv1alpha1.ProgressiveRolloutStage) []string {
```

Comment:
Is this just a filter?
why is it returning a `[]string` rather than `[]Application`?

Reply:
Because I don't know how the implementation of sync will look like just yet :)
If we used the CLI for example, we just need the app name as string.

Reply:
we can still have a `GetNameFromApplications` later if we need to?

Reply:
Fair point, I'll fix it to return apps and not string